### PR TITLE
Fix check-packages on CI broken due to enzyme removal

### PR DESCRIPTION
**Pull request by brentvatne**
# Why

Fix tests that I broke when removing `expo-module-scripts/enzyme`

# How

Use testing-library instead

# Test Plan

Run CI

### CODE REVIEW

File: packages/expo-linear-gradient/src/__tests__/LinearGradient-test.web.tsx [hunk 1]
```diff
@@ -1,31 +1,58 @@
...
-const getStyleProp = (component, prop) => StyleSheet.flatten(component.prop('style'))[prop];
-
 it(`renders a multi-color gradient background with alpha`, () => {
   const colors = ['cyan', '#ff00ff', 'rgba(0,0,0,0)', 'rgba(0,255,255,0.5)'];
-  const component = mount(<LinearGradient colors={colors} />);
-  const backgroundImage = getStyleProp(component.find('View'), 'backgroundImage');
+  const component = render(<LinearGradient colors={colors} testID="gradient" />);
+  const view = getByTestId(component.container, 'gradient');
+
+  // TODO: figure out how to test this - not sure what the right way is to read
```
Comment (brentvatne) [inline packages/expo-linear-gradient/src/__tests__/LinearGradient-test.web.tsx:15]: will follow up: https://linear.app/expo/issue/ENG-6854/fix-expo-linear-gradient-testing-library-tests

File: packages/expo-checkbox/src/ExpoCheckbox.web.tsx [hunk 2]
```diff
@@ -42,7 +42,7 @@ const ExpoCheckbox: CheckboxComponent = (props) => {
   );
 
   const nativeControl = createElement('input', {
-    accessibilityState: { disabled, checked: value },
+    accessibilitystate: { disabled, checked: value },
```
Comment (tsapeta) [inline packages/expo-checkbox/src/ExpoCheckbox.web.tsx:45]: 👀

**Review by tsapeta — APPROVED** [review]
Thanks for fixing that! 🙏

File: packages/expo-checkbox/src/ExpoCheckbox.web.tsx (hunk 2 above)
Comment (brentvatne) [reply packages/expo-checkbox/src/ExpoCheckbox.web.tsx:45]: i was told by react that this was needed but let me double check this

Comment (brentvatne) [reply packages/expo-checkbox/src/ExpoCheckbox.web.tsx:45]: image

😰

Comment (tsapeta) [reply packages/expo-checkbox/src/ExpoCheckbox.web.tsx:45]: Cool, thanks for double-checking 😉

Comment (brentvatne) [reply packages/expo-checkbox/src/ExpoCheckbox.web.tsx]: proper fix: https://github.com/expo/expo/pull/19791/commits/9c1276c3e0583fe8ef635b3c951f923ed11a5976

**Pull request merged**